aten/src/ATen/native/Loss.cpp

```diff
@@ -148,4 +148,36 @@ Tensor poisson_nll_loss(const Tensor& input, const Tensor& target, const bool lo
 
     return apply_loss_reduction(loss, reduction);
 }
+
+Tensor l1_loss(const Tensor& input, const Tensor& target, int64_t reduction) {
+  auto loss = input.sub(target).abs_();
```

Comment:
This should cause performance regressions for smaller tensors. The original kernel in THNN was fused; now it is not fused. I think a better way to do this would be to port the entire function to use TensorIterator.

Reply:
@zou3519 I am not really sure if the extra complexity is worth this. The super-simple port with basic tensor operations already significantly improves the performance compared to the original fused implementation. At least for Tensors > 128x100. Before going the route of using tensor-iterator I would suggest to first perform further benchmarks with smaller tensors. e.g. see my test for soft_margin_loss  [here](https://github.com/pytorch/pytorch/pull/27673) which showed that a port even for 1000 element-tensors is faster.

Reply:
@andreaskoepf, there will have a big  performance degression if not using fused ops on GPU deveice for mse_loos which will call many other ops, please see https://github.com/pytorch/pytorch/pull/26529, for l1_loss, there only have a little perfromance degression  for the backward case, perhaps we don't need fused them, wait @VitalyFedyunin's response.

Reply:
I think we can live with tiny regression introduced by additional dispatch as we are planning to speed it up soon.